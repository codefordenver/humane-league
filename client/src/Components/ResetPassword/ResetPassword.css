.ForgotPassword {
  display: grid;
  grid-template-columns: 100px 1fr;
  width: 600px;
  margin: 0px auto;
  grid-gap: 10px;
  background-color: #ffffffe3;
  padding: 10px; }
  .ForgotPassword h1 {
    grid-column: span 2;
    text-align: center;
    margin: 0px; }
  .ForgotPassword h5 {
    grid-column: span 2;
    text-align: center; }
  .ForgotPassword label {
    display: flex;
    align-items: center; }
  .ForgotPassword button {
    grid-column: span 2;
    height: 70px;
    display: flex;
    width: 250px;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    border: none;
    background: #1CB841;
    color: white;
    cursor: pointer;
    margin: 0px auto;
    transition: 0.3s;
    font-size: 20px;
    padding: 10px; }
    .ForgotPassword button:hover {
      color: white;
      background-color: #323232; }
    .ForgotPassword button:disabled {
      background: #989696;
      color: black; }
  .ForgotPassword input {
    height: 50px;
    font-size: 24px;
    padding-left: 10px;
    outline: none; }
  .ForgotPassword .confirmWrapper {
    position: relative; }
    .ForgotPassword .confirmWrapper input {
      width: 100%; }
    .ForgotPassword .confirmWrapper span {
      position: absolute;
      display: flex;
      justify-content: center;
      align-items: center;
      z-index: 10;
      color: black;
      right: 0px;
      top: 0px;
      height: 100%;
      min-width: 50px; }
