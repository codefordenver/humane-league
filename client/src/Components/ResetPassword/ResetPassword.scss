.ForgotPassword {
  display: grid;
  grid-template-columns: 100px 1fr;
  width: 600px;
  margin: 0px auto;
  grid-gap: 10px;
  background-color: #ffffffe3;
  padding: 10px;

  h1 {
    grid-column: span 2;
    text-align: center;
    margin: 0px;
  }

  h5 {
    grid-column: span 2;
    text-align: center;
  }

  label {
    display: flex;
    align-items: center;
  }

  button {
    grid-column: span 2;
    height: 70px;
    display: flex;
    width: 250px;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    border: none;
    background: #1CB841;
    color: white;
    cursor: pointer;
    margin: 0px auto;
    transition: 0.3s;
    font-size: 20px;
    padding: 10px;
    &:hover {
      color: white;
      background-color: rgb(50, 50, 50);
    }
    &:disabled {
      background: rgb(152, 150, 150);
      color: black;
    }
  }

  input {
    height: 50px;
    font-size: 24px;
    padding-left: 10px;
    outline: none;
  }

  .confirmWrapper {
    position: relative;
    input {
      width: 100%;
    }

    span {
      position: absolute;
      display: flex;
      justify-content: center;
      align-items: center;
      z-index: 10;
      color: black;
      right: 0px;
      top: 0px;
      height: 100%;
      min-width: 50px;
    }

  }
}