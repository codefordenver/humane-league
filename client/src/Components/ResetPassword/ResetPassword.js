import React, { Component } from 'react';
import './ResetPassword.css';
import firebase from '../../firebase';


class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      validLink: false,
      password: '',
      confirmPass: '',
      disableButton: true,
      passMatchStatus: null
    };
  }

  componentDidMount() {
    console.log(this.props);
    // firebase.auth().verifyPasswordResetCode
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });

    if (
      this.password.value.length > 5 &&
      this.password.value === this.confirmPass.value
    ) {
      this.setState({disableButton: false, passMatchStatus: '✅' });
    } else {
      this.setState({disableButton: true, passMatchStatus: '❌' }); 
    }

  }

  render() {
    return (
      <div className="ForgotPassword">
        <h1>Reset Password</h1>
        <h5>Passwords must be at least 6 characters long.</h5>
        <label>Password: </label>
        <input ref={(input) => this.password = input} placeholder="Password" name="password" value={this.state.password} onChange={this.handleChange} type="text" />
        <label>Confirm Password: </label>
        <div className="confirmWrapper">
          <input 
            ref={(input) => { this.confirmPass = input }} 
            name="confirmPass" 
            onChange={this.handleChange} 
            value={this.state.confirmPass}
            type="password" 
            placeholder="Confirm Password" />
          <span>{this.state.passMatchStatus}</span>
        </div> 
        <button disabled={this.state.disableButton}>Change Password</button>
      </div>
    )
  }
}
export default ForgotPassword;