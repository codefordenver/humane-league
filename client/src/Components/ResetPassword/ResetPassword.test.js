import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ResetPassword from './ResetPassword';

describe('ResetPassword component tests', () => {
  let renderedResetPassword;
  let defaultState; 
  
  beforeEach(() => {
    renderedResetPassword = shallow( <ResetPassword /> );
    defaultState ={
      validLink: false,
      password: '',
      confirmPass: '',
      disableButton: true,
      passMatchStatus: null
    }
  })
  it('renders without crashing', () => {
    expect(renderedResetPassword).toBeDefined();
    expect(renderedResetPassword.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedResetPassword).toMatchSnapshot();
  })
})