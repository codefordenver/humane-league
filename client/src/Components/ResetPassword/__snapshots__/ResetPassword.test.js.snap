// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Reset Password
</h1>,
        <h5>
          Passwords must be at least 6 characters long.
</h5>,
        <label>
          Password: 
</label>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="text"
          value=""
/>,
        <label>
          Confirm Password: 
</label>,
        <div
          className="confirmWrapper"
>
          <input
                    name="confirmPass"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
          <span />
</div>,
        <button
          disabled={true}
>
          Change Password
</button>,
      ],
      "className": "ForgotPassword",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Password",
        },
        "ref": null,
        "rendered": "Reset Password",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Passwords must be at least 6 characters long.",
        },
        "ref": null,
        "rendered": "Passwords must be at least 6 characters long.",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password: ",
        },
        "ref": null,
        "rendered": "Password: ",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "text",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Confirm Password: ",
        },
        "ref": null,
        "rendered": "Confirm Password: ",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="confirmPass"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            <span />,
          ],
          "className": "confirmWrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "confirmPass",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Change Password",
          "disabled": true,
        },
        "ref": null,
        "rendered": "Change Password",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Reset Password
</h1>,
          <h5>
            Passwords must be at least 6 characters long.
</h5>,
          <label>
            Password: 
</label>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="text"
            value=""
/>,
          <label>
            Confirm Password: 
</label>,
          <div
            className="confirmWrapper"
>
            <input
                        name="confirmPass"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        type="password"
                        value=""
            />
            <span />
</div>,
          <button
            disabled={true}
>
            Change Password
</button>,
        ],
        "className": "ForgotPassword",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Password",
          },
          "ref": null,
          "rendered": "Reset Password",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Passwords must be at least 6 characters long.",
          },
          "ref": null,
          "rendered": "Passwords must be at least 6 characters long.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password: ",
          },
          "ref": null,
          "rendered": "Password: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirm Password: ",
          },
          "ref": null,
          "rendered": "Confirm Password: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="confirmPass"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
/>,
              <span />,
            ],
            "className": "confirmWrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "confirmPass",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Change Password",
            "disabled": true,
          },
          "ref": null,
          "rendered": "Change Password",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
